compiler/vm.nim

```diff
@@ -2077,12 +2117,15 @@ proc rawExecute(c: PCtx, start: int, tos: PStackFrame): TFullReg =
         if regs[ra].node.strVal.isNil: regs[ra].node.strVal = newStringOfCap(1000)
       storeAny(regs[ra].node.strVal, typ, regs[rb].regToNode, c.config)
 
+    profiler.leave(c)
+
     inc pc
 
 proc execute(c: PCtx, start: int): PNode =
   var tos = PStackFrame(prc: nil, comesFrom: 0, next: nil)
   newSeq(tos.slots, c.prc.maxSlots)
   result = rawExecute(c, start, tos).regToNode
+  c.dumpProfile()
```

Comment:
instead, dump once at the end, right before `rawMessage(conf, hintSuccessX...`
(also makes it easy to modify dumping logic to instead output to some json)

Reply:
Hmm,  can't reach the vm's `PCtx` from there. I was looking for a single place to dump, but I couldn't find the proper "all is done" in the vm code...

Reply:
see https://github.com/nim-lang/Nim/pull/14833#discussion_r446568040 for a solution

Reply:
Done